src/git_changelog/build.py

```diff
@@ -137,6 +136,21 @@ def is_minor(self) -> bool:
         """
         return bool(self.tag.split(".", 2)[2])
 
+    def add_commit(self, commit: Commit) -> None:
+        """Adds the given commit and assigns it a section based on the chosen commit convention.
+
+        Arguments:
+            commit: The git commit.
+        """
+        self.commits.append(commit)
+        commit.version = self.tag
+        _type = commit.convention["type"]
+        if "type" in commit.convention and _type not in self.sections_dict:
+            section = Section(section_type=_type)
+            self.sections_list.append(section)
+            self.sections_dict[_type] = section
+        self.sections_dict[_type].commits.append(commit)
```

Comment:
If `"type"` is not in `commit.convention`, `commit.convention["type"]` would fail with KeyError.

```suggestion
        commit_type = commit.convention.get("type")
        if commit_type not in self.sections_dict:
            section = Section(section_type=commit_type)
            self.sections_list.append(section)
            self.sections_dict[commit_type] = section
        self.sections_dict[commit_type].commits.append(commit)
```